packages/scripts/src/generate-milestones.ts

```diff
@@ -148,23 +165,73 @@ const checkToken = async (start: Date) => {
   }
 };
 
+const COUNTRY_OF_EDUCATIONS = {
+  us: 'us',
+  uk: 'uk',
+  ie: 'ireland',
+  in: 'india',
+  au: 'australia',
+  ph: 'philippines',
+  ng: 'nigeria',
+  jm: 'jamaica',
+  ke: 'kenya',
+  ca: 'canada',
+  'n/a': 'n/a',
+};
+
+const getNewApplicant = (from: string, to: string): IENApplicantCreateUpdateDTO => {
+  const first_name = faker.name.firstName();
+  const last_name = faker.name.lastName();
+  const registration = faker.date.between(from, to);
+  return {
+    applicant_id: faker.datatype.uuid(),
+    first_name,
+    last_name,
+    email_address: faker.internet.email(first_name, last_name),
+    phone_number: faker.phone.number('###-###-####'),
+    registration_date: dayjs(registration).format('YYYY-MM-DD'),
+    nursing_educations: [],
+    country_of_citizenship: [faker.helpers.arrayElement(Object.keys(COUNTRY_OF_EDUCATIONS))],
+    country_of_residence: faker.helpers.arrayElement(Object.keys(COUNTRY_OF_EDUCATIONS)),
+    pr_status: 'No status',
+    is_open: true,
+  };
+};
+
+const addApplicant = async (from: string, to: string): Promise<ApplicantRO> => {
+  const applicant = getNewApplicant(from, to);
+  const { data } = await axios.post<{ data: ApplicantRO }>(`/ien`, applicant);
+  return data?.data;
+};
+
 axios.defaults.baseURL = 'http://localhost:4000/api/v1';
 
 setToken()
   .then(async () => {
-    const [applicants, count] = await getApplicants();
+    const [applicants, initialCount] = await getApplicants();
+
+    console.log(`${initialCount} applicants found`);
+
+    while (applicants.length < APPLICANT_COUNT) {
+      const applicant = await addApplicant('2021-01-01', '2022-01-01');
+      applicants.push(applicant);
+    }
+    const count = applicants.length;
+
+    console.log(`${count - initialCount} applicants added`);
+
+    const applicantsWithoutMilestones = applicants.filter((a: any) => !a.status);
+
+    console.log(`${applicantsWithoutMilestones.length} applicants have no milestone.`);
+
     const statuses = await getMilestones();
-    for (let i = 0; i < count; i++) {
+    for (let i = 0; i < applicantsWithoutMilestones.length; i++) {
       await checkToken(start);
 
-      const applicant = applicants[i];
-      if (dayjs(applicant.registration_date).isAfter('2022-06-01')) {
-        console.log(`skip applicant registered at ${applicant.registration_date}`);
-        continue;
-      }
+      const applicant = applicantsWithoutMilestones[i];
 
-      await generateMilestones(applicant.id, statuses);
-      console.log(`${i}/${count} applicant processed <- ${applicant.id}`);
+      const numOfMilestones = await generateMilestones(applicant.id, statuses);
```

Comment:
`addApplicant` returns an `ApplicantRO` couldn't we just pass the applicant instead of the id and having to use `await getApplicant(id)` in `generateMilestones`?

Reply:
The applicants list contains the result of `getApplicants` that do not have milestones.

Reply:
yeah, that's what I was asking.  Was curious if `applicantsWithoutMilestones` is just a filtered list of `ApplicantRO` then we could just pass the applicant object instead of the id? As the `generateMilestones` fn is calling `getApplicant(id)`, just wondering if it's returning data we already had before calling

Reply:
You're right if we generate milestones only for the newly added applicants.